sklearn/neural_network/tests/test_mlp.py

```diff
@@ -966,3 +967,47 @@ def test_mlp_partial_fit_after_fit(MLPEstimator):
     msg = "partial_fit does not support early_stopping=True"
     with pytest.raises(ValueError, match=msg):
         mlp.partial_fit(X_iris, y_iris)
+
+
+@pytest.mark.parametrize("weighted_class", [i for i in range(3)])
+def test_mlp_classifier_with_sample_weights(weighted_class):
```

Comment:
Also the regressor also needs to be tested.

For the classifier, there needs to be a test with two classes so that `binary_log_loss` is tested with sample weights.